src/programs.rkt

```diff
@@ -306,15 +300,28 @@
 (define (expand-parametric-reverse expr repr)
   (define ->bf (representation-repr->bf repr))
   (match expr
+    [(list 'if cond ift iff)
+     (define cond* (expand-parametric-reverse cond repr))
+     (define ift* (expand-parametric-reverse ift repr))
+     (define iff* (expand-parametric-reverse iff repr))
+     (list 'if cond* ift* iff*)]
+    [(list (? (compose (curry regexp-match? #rx"[A-Za-z0-9_]+(->)[A-Za-z0-9_]+") symbol->string) op) 
+           body) ; conversion (e.g. posit16->f64)
+     (define-values (iprec oprec)
+       (let ([split (string-split (symbol->string op) "->")])
+         (values (first split) (last split))))
+     (define repr* (get-representation (string->symbol oprec)))
+     (define body* (expand-parametric-reverse body repr*))
+     `(! :precision ,oprec ,body*)]
```

Comment:
OMG... This is... oof. Let's meet and try to think of something better.

Reply:
Haha. This was just temporary. It's pretty bad.